react/components/DatePicker/index.js

```diff
@@ -153,6 +191,8 @@ DatePicker.propTypes = {
   readOnly: PropTypes.bool,
   /** Spec attribute  */
   required: PropTypes.bool,
+  /** Whether or not the popper should be aligned along the right side of the input */
+  rightAligned: PropTypes.bool,
```

Comment:
this prop could follow the standard set by `ActionMenu`, i.e. `align: left|right`